CRM/Utils/SQL/TempTable.php

```diff
@@ -287,4 +294,16 @@ public function setUtf8($value = TRUE) {
     return $this;
   }
 
+  /**
+   * Set table engine to MEMORY.
+   *
+   * @param bool $value
+   *
+   * @return $this
+   */
+  public function setMemory($value = TRUE) {
```

Comment:
Aaah. I like the idea of using MEMORY more, but I've never used it because of the limitations on `text`/`blob` columns. This exposes it as an opt-in... so it would avoid that limitation because it's only activated in use-cases where the developer can reasonably predict that the temp-table will be compatible with MEMORY?

Reply:
Yep. And CiviCRM uses it quite a bit already; just grep the codebase for HEAP and MEMORY.